src/utils/Firebase/firebase.js

```diff
@@ -106,8 +106,19 @@ class Firebase {
 
         return Promise.all(customLessonRefs)
           .then(customLessonDocs => {
-            const customLessons = customLessonDocs.map(doc => doc.data());
-            return customLessons;
+            let customLessons = customLessonDocs.map(doc => doc.data());
+            let assignments = [];
+
+            // Get due dates for assignments
+            for (let customLesson in customLessons) {
+              for (let dueDate in customLesson.dueDates[deploymentAccountId]) {
+                let lessonCopy = JSON.parse(JSON.stringify(customLesson));
```

Comment:
@Lam7150

Reply:
@Lam7150 ?

Reply:
Yes! This is to make a deep copy of the lesson, so that the original object is not affected -- it was the best solution I found online, the syntax is definitely weird though